File: ledger/ledger-consensual/src/main/java/net/corda/v5/ledger/consensual/Party.java
package net.corda.v5.ledger.consensual;

import net.corda.v5.base.annotations.DoNotImplement;
import net.corda.v5.base.types.MemberX500Name;
import org.jetbrains.annotations.NotNull;

import java.security.PublicKey;

/**
 * Representation of a Party.
 *
 *
 * Since used in {@link ConsensualState} interface, their implementors will depend on it.
 * To make those implementations serializable, we use a custom serializer in the background, so
 * the normal @CordaSerializable cannot be used here.
 */
@DoNotImplement
public interface Party {
    /**
     * @return name X500 Name of the party.
     */
    @NotNull
    MemberX500Name getName();

    /**
     * @return owningKey The {@link PublicKey} owned by the party.
     */
    @NotNull
    PublicKey getOwningKey();
}

File: ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/Party.kt
package net.corda.v5.ledger.consensual

import net.corda.v5.base.annotations.CordaSerializable
import net.corda.v5.base.annotations.DoNotImplement
import net.corda.v5.base.types.MemberX500Name
import net.corda.v5.crypto.toStringShort
import java.security.PublicKey

/**
 * Representation of a Party.
 */
@DoNotImplement
@CordaSerializable
interface Party {
    /**
     * @property name X500 Name of the party.
     */
    val name: MemberX500Name

    /**
     * @property owningKey The [PublicKey] owned by the party.
     */
    val owningKey: PublicKey

    /**
     * Concise, readable String representation of a Party.
     */
    val description: String get() = "$name (owningKey = ${owningKey.toStringShort()})"
}